### PR TITLE
[2.1 backport] Port the release script from master

### PR DESCRIPTION
**Pull request by kit-ty-kate**
This would allow us to use the release script as-is from the tag instead of from master. This would help reproducibility

Related to https://github.com/ocaml/opam/issues/5958
Backports https://github.com/ocaml/opam/pull/6084

### CODE REVIEW

File: release/readme.md [hunk 1]
```diff
@@ -1,29 +1,54 @@
...
 [ once bump version & changes PRs merged ]
-* tag the release (git tag -a 2.2.0; git push origin 2.2.0)
-* /!\ Once the tag pushed, it can be updated only in case of severe issue
+* tag the release (git tag -am 2.2.0 2.2.0; git push origin 2.2.0)
+* /!\ Once the tag pushed, it can be updated [different commit] only in case of severe issue
 * create a release (or prerelease if intermediate release) draft on github based on your tag (https://github.com/ocaml/opam/releases/new)
-* generate opam artifacts, using `shell/release.sh`, it requires to have Docker install with several remotes, the different arches
-* add releases notes (content of `master_changes.md`) in the release
-* upload signature of artefacts
+* fetch locally the tag
```
Comment (rjbou) [inline release/readme.md:17]: It's missing the readme update, otherwise, lgtm

Comment (kit-ty-kate) [reply release/readme.md]: done

**Review by rjbou — APPROVED** [review]
Thanks!

**Comment by rjbou** [comment]
OCaml bench test is broken

**Pull request merged**